### PR TITLE
feat: new job to clean up prometheus gauges of deleted checks.

### PR DESCRIPTION
**Pull request by adityathebe**
Resolves: https://github.com/flanksource/canary-checker/issues/1073

### CODE REVIEW

File: pkg/jobs/canary/prometheus_cleanup.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+			checkIDs = append(checkIDs, string(canaryName))
+		}
+	}
+
+	deletedChecks, err := db.FindDeletedChecksByIDs(ctx, checkIDs)
+	if err != nil {
+		logger.Errorf("Error finding deleted checks: %v", err)
+		return
+	}
+	logger.Infof("Found %d/%d deleted checks", len(deletedChecks), len(checkIDs))
```
Comment (yashmehrotra) [inline pkg/jobs/canary/prometheus_cleanup.go:52]: (nit) Can be Debugf

File: pkg/jobs/jobs.go [hunk 2]
```diff
@@ -3,8 +3,10 @@ package jobs
 import (
 	v1 "github.com/flanksource/canary-checker/api/v1"
 	"github.com/flanksource/canary-checker/pkg/db"
+	"github.com/flanksource/canary-checker/pkg/jobs/canary"
```
Comment (yashmehrotra) [inline pkg/jobs/jobs.go:6]: duplicate import

File: pkg/jobs/jobs.go [hunk 3]
```diff
@@ -54,6 +56,15 @@ func Start() {
 	if _, err := ScheduleFunc(v1.CheckStatusesAggregate1dSchedule, db.AggregateCheckStatuses1d); err != nil {
 		logger.Errorf("Failed to schedule check statuses aggregator 1d: %v", err)
 	}
+
+	if runner.Prometheus != nil {
+		if _, err := ScheduleFunc(v1.PrometheusGaugeCleanupSchedule, canary.CleanUpPrometheusGauges); err != nil {
+			logger.Errorf("Failed to schedule prometheus gauge cleanup job: %v", err)
+		}
+
+		canary.CleanUpPrometheusGauges()
```
Comment (yashmehrotra) [inline pkg/jobs/jobs.go:65]: canaryJobs

File: pkg/db/canary.go [hunk 4]
```diff
@@ -233,6 +234,12 @@ func FindCheck(canary pkg.Canary, name string) (*pkg.Check, error) {
 	return &model, nil
 }
 
+func FindDeletedChecksSince(ctx context.Context, since time.Time) ([]models.Check, error) {
+	var checks []models.Check
+	err := Gorm.Where("deleted_at > ?", since).Find(&checks).Error
```
Comment (yashmehrotra) [inline pkg/db/canary.go:239]: Only query IDs

Comment (moshloop) [reply pkg/db/canary.go:239]: I also thing we should only return ids

**Pull request merged**